packages/react-core/src/components/Page/Page.tsx

```diff
@@ -265,30 +266,21 @@ class Page extends React.Component<PageProps, PageState> {
     let nav = null;
     if (horizontalSubnav && isHorizontalSubnavWidthLimited) {
       nav = (
-        <div className={css(styles.pageMainNav, styles.modifiers.limitWidth)}>
-          <div className={css(styles.pageMainBody)}>{horizontalSubnav}</div>
+        <div className={css(styles.pageMainSubnav, styles.modifiers.limitWidth)}>
+          <PageBody>{horizontalSubnav}</PageBody>
         </div>
       );
     } else if (horizontalSubnav) {
-      nav = <div className={css(styles.pageMainNav)}>{horizontalSubnav}</div>;
+      nav = <div className={css(styles.pageMainSubnav)}>{horizontalSubnav}</div>;
     }
```

Comment:
Since [`<PageBody>` should wrap the children of all sections except for `page__main-group`](https://github.com/patternfly/patternfly/issues/6801), I think you could update this so `<PageBody>` is there with and without `isHorizontalSubnavWidthLimited`, and we just dynamically render `styles.modifiers.limitWidth` based on whether that prop is true or false.